File: dynamicProgramming/Coin Change.cpp
class Solution
{
  public:
    long long int count( int S[], int m, int n )
    {
       
        //code here.
        long long int table[n+1][m+1];
        for (int j =0;j<m+1;j++){
            table[0][j] = 1;
        }
        for(int i = 0;i<n+1;i++){
            table[i][0] = 0;
        }
        for (int i = 1;i<n+1;i++){
            for(int j=1;j<m+1;j++){
                table[i][j] = table[i][j-1];
                if(S[j-1]<=i){
                    table[i][j]+=table[i-S[j-1]][j];
                }
            }
        }
        return table[n][m];
    }
};

File: dynamicProgramming/Coin Change.py
class Solution:
    def count(self, S, m, n): 
        # code here 
        dp = [[0 for j in range(m+1)] for i in range(n+1)]
        for j in range(m+1):
            dp[0][j] = 1
        for i in range(1,n+1):
            for j in range(1,m+1):
                dp[i][j] = dp[i][j-1]
                if S[j-1]<=i:
                    dp[i][j]+=dp[i-S[j-1]][j]
        return dp[n][m]
